### ISSUE TITLE
Dropout(0.1) sets the layer's name to 0.1 and leaves p at 0.5

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor-ml#121]
On `Dropout`, the pads, the pools and the norms, `name` is the first positional parameter and is optional, so the torch and keras spelling puts the hyperparameter into the name field. The value is truthy, formats fine into every f-string, and the hyperparameter keeps its default. Nothing raises. What makes it worse is that `Linear`, `Conv2D`, `Embedding` and `MultiheadAttention` *require* `name` positionally and so fail loudly — a user learns the convention from the layers that shout and then gets burned silently by the ones that do not.

```python
from pytensor_ml.layers import BatchNorm, Dropout, MaxPool2D, ZeroPad2D

print(Dropout(0.1).name, Dropout(0.1).p)              # 0.1 0.5   <- half the units dropped
print(MaxPool2D(3).name, MaxPool2D(3).kernel_size)    # 3 (2, 2)
print(ZeroPad2D(1).name, ZeroPad2D(1).padding)        # 1 ((0, 0), (0, 0))   <- a no-op
print(BatchNorm(32).name, BatchNorm(32).n_in)         # 32 None
```

Rejecting a non-string first positional argument on these layers would turn all four into an immediate error naming the parameter the caller meant.
